ansible/playbooks/roles/image_registry/tasks/main.yml

```diff
@@ -36,16 +36,18 @@
       when:
         - check_docker.rc == 0
 
-    - name: Check if previous version registry is running and kill it if needed
+    - name: Check if previous version registry is running and stop it if needed
       become: true
       block:
         - name: Get current running registry container
           command: "{% raw %}docker ps --format '{{ .Image }}' -f 'name=registry' -f 'status=running'{% endraw %}"
           register: running_registry_images
 
-        - name: Kill previous version running registry
-          command: docker kill -q -f '{{ running_registry_images.stdout }}'
-          when: specification.registry_image.name != '{{ running_registry_images.stdout }}'
+        - name: Stop previous version running registry
+          command: docker stop '{{ running_registry_images.stdout }}'
```

Comment:
It won't work anyway. `running_registry_images.stdout` contains the name of the image, and the `docker stop` command should be executed against the container, not the image.
You should also probably register the output from `docker ps --format '{{ .Names }}' -f 'name=registry' -f 'status=running'` and then use it in the `docker stop` command. You should also probably register the output from `docker ps --format '{.Names}}' -f 'name = register' -f 'status = running'` and use it in the` docker stop` command. Or some other clever way. :)